identity/app/idapiclient/IdApiClient.scala

```diff
@@ -89,11 +89,12 @@ class IdApiClient(
     response map extract[Boolean](jsonField("passwordExists"))
   }
 
-  def updatePassword(pwdUpdate: PasswordUpdate, auth: Auth, trackingData: TrackingData ): Future[Response[Unit]] = {
+  def updatePassword(pwdUpdate: PasswordUpdate, auth: Auth, trackingData: TrackingData ): Future[Response[CookiesResponse]] = {
     val apiPath = urlJoin("user", "password")
     val body = write(pwdUpdate)
-    val response = post(apiPath, Some(auth), Some(trackingData), Some(body))
-    response map extractUnit
+    val headers = buildHeaders(Some(auth), extra = xForwardedForHeader(trackingData))
+    val response = httpClient.POST(apiUrl(apiPath), Some(body), clientAuth.parameters, headers)
+    response map extract(jsonField("cookies"))
```

Comment:
I think it's preferable to explicitly specify the type you are extracting using the type parameter, rather than letting the compiler work it out from the return type i.e. `extract[CookiesResponse](...)`.

Reply:
I won't try and push for not using the space syntax here, since most methods in this class use it 😅